### DISCUSSION TITLE
Regex Manager for Go version fails to update dependency

### DISCUSSION THREAD

**Discussion by juanjjaramillo** [renovatebot/renovate#25150] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I wrote the following regex manager to update a Go dependency:
```json
    {
      "description": "Update Go version",
      "customType": "regex",
      "fileMatch": [
        "(^|/)ci\.ya?ml$",
      ],
      "matchStrings": [
        "\\sGO_VERSION: '(?<currentValue>.*)'\\s"
      ],
      "depNameTemplate": "go",
      "datasourceTemplate": "golang-version",
      "versioningTemplate": "go-mod-directive",
    },
```

The `fileMatch` file contains the following text:
```yaml
env:
  GO_VERSION: '1.20'
```
However, dependency does not get updated as relevant logs show:
```json
{
  "deps": [
    {
      "depName": "go",
      "currentValue": "1.20",
      "datasource": "golang-version",
      "versioning": "go-mod-directive",
      "replaceString": " GO_VERSION: '1.20'\n",
      "updates": [],
      "packageName": "go",
      "warnings": [],
      "sourceUrl": "https://github.com/golang/go",
      "registryUrl": "https://raw.githubusercontent.com/golang/website",
      "homepage": "https://go.dev/",
      "currentVersion": "1.21.3"
    }
  ],
  "matchStrings": [
    "\\sGO_VERSION: '(?<currentValue>.*)'\\s"
  ],
  "depNameTemplate": "go",
  "datasourceTemplate": "golang-version",
  "versioningTemplate": "go-mod-directive",
  "packageFile": ".github/workflows/ci.yml"
}
```

### Relevant debug logs

<details><summary>Logs</summary>

```json
{
  "deps": [
    {
      "depName": "go",
      "currentValue": "1.20",
      "datasource": "golang-version",
      "versioning": "go-mod-directive",
      "replaceString": " GO_VERSION: '1.20'\n",
      "updates": [],
      "packageName": "go",
      "warnings": [],
      "sourceUrl": "https://github.com/golang/go",
      "registryUrl": "https://raw.githubusercontent.com/golang/website",
      "homepage": "https://go.dev/",
      "currentVersion": "1.21.3"
    }
  ],
  "matchStrings": [
    "\\sGO_VERSION: '(?<currentValue>.*)'\\s"
  ],
  "depNameTemplate": "go",
  "datasourceTemplate": "golang-version",
  "versioningTemplate": "go-mod-directive",
  "packageFile": ".github/workflows/ci.yml"
}
```

</details>

**Comment by rarkins** [comment]
Please update the title and description to reflect your current question

**Reply by juanjjaramillo** [reply]
@rarkins could you please elaborate? I think the title is accurate, and the description covers all the requested fields. In section `What you tried so far.` I only cover alternatives I tried, but that is not the solution to the original question, it is only a dead-end

**Reply by rarkins** [reply]
It no longer failed with a skipReason once you applied the right versioning. Unless you want me to mark your own comment as the correct answer and close this, please rewrite your question

**Reply by juanjjaramillo** [reply]
@rarkins please let me know if the new title and description work for you. I would appreciate you helping us out finding why the regex manager is not working

**Reply by rarkins** [reply, marked as answer]
The line `"currentVersion": "1.21.3"` looks like a hint. The Go mod directive does not mean "exactly X", it means "X or greater". 1.21.3 is the highest version which satisfies that.

If you configure rangeStrategy=bump then I think you'll find that you get an update proposed, because that range strategy tells Renovate that you want the value bumped to the latest version each time. Please let me know if that works
